app/components/channel_modal.js

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2016 Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {Component, PropTypes} from 'react';
+import {Animated, Dimensions} from 'react-native';
+
+const AnimatedView = Animated.View;
+const {width: deviceWidth, height: deviceHeight} = Dimensions.get('window');
+
+export default class ChannelModel extends Component {
+    static propTypes = {
+        children: PropTypes.node,
+        duration: PropTypes.number,
+        topOffset: PropTypes.number,
+        theme: React.PropTypes.object.isRequired,
+        visible: PropTypes.bool.isRequired
+    }
+
+    static defaultProps = {
+        duration: 400,
+        topOffset: 0,
+        visible: false
+    }
+
+    constructor(props) {
```

Comment:
I prefer not to use the constructor unnecessarily. Instead I would set initial state like this:
```
export default class ChannelModal extends Component {
  static propTypes...
  static defaultProps...
  
  state = {
    top: new Animated.Value(deviceHeight)
  };
}
```